storm-core/src/jvm/backtype/storm/utils/DisruptorQueue.java

```diff
@@ -138,7 +150,21 @@ private void consumeBatchToCursor(long cursor, EventHandler<Object> handler) {
         //TODO: only set this if the consumer cursor has changed?
         _consumer.set(cursor);
     }
-    
+
+    public void registerBackpressureCallback(DisruptorBackpressureCallback cb) {
+        this._cb = cb;
+    }
+
+    static public void notifyBackpressureChecker(Object trigger) {
```

Comment:
I don't think this code really belongs here.  It would make more since to move it to the WorkerBackpressureThread itself.